packages/core/content-type-builder/admin/src/components/FormModal/forms/utils/createCollectionName.ts

```diff
@@ -0,0 +1,8 @@
+import { snakeCase } from 'lodash/fp';
+import pluralize from 'pluralize';
+
+const createComponentCollectionName = (name: string, category: string) => {
```

Comment:
It looks like this comes from [here](https://github.com/strapi/strapi/blob/434c4b303dba208ca564ed5fd1d6621931082d43/packages/core/content-type-builder/server/src/services/schema-builder/component-builder.ts#L40) originally. But I'm worried if we start copy/pasting the logic around, it's going to get very messy, especially since we're planning some big changes around table naming.

Could you look into refactoring this (and the code I linked) into a utility method for components to generate a collection name?

And if it doesn't seem feasible right now, at least please add a comment both places to reference each other so that if one gets updated they both do because that's probably happening very soon.

Reply:
I decided to add some comments, importing `@strapi/utils` to admin seems to make the admin build process to break, some webpack stuff I don't want dig into now

Reply:
Yeah, we don't want to introduce a new circular dependancy. We will clean it up when we refactor the db identifier naming methods